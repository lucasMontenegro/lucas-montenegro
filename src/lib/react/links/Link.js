import React from "react"
import { Link as RouterLink } from "react-router-dom"
import MuiLink from "@material-ui/core/Link"
const ForwardLinkRef = React.forwardRef(function ForwardLinkRef (props, ref) {
  return <RouterLink {...props} innerRef={ref} />
})
export default function Link (props) {
  return <MuiLink {...props} component={ForwardLinkRef} />
}