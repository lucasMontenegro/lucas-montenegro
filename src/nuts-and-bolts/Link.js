import React from 'react';
import { Link as RouterLink } from 'react-router-dom';
import MaterialLink from '@material-ui/core/Link';
import clone from 'lodash/clone';

class Link extends React.Component {
  Link = React.forwardRef((itemProps, ref) => (
    <RouterLink {...itemProps} to={this.props.to} innerRef={ref} />
  ));

  render() {
    const props = clone(this.props);
    props.component = this.Link;
    delete props.to;
    return <MaterialLink {...props} />;
  }
}

export default Link;
