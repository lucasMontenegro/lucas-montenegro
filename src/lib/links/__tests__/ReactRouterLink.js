import React from "react"
import Link from "lib/links/ReactRouterLink"
describe(`lib/links/ReactRouterLink`, () => {
  it(`should use the right dependency versions`, () => {
    expect(jestUtils.getDependencies([`react`, `react-router-dom`, `@material-ui/core`]))
      .toMatchSnapshot()
  })
  it(`should run`, () => {
    <Link to="/somewhere">somewhere</Link>
  })
})