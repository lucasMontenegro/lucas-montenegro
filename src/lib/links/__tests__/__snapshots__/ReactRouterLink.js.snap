// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/links/ReactRouterLink should use the right dependency versions 1`] = `
Object {
  "@material-ui/core": "^4.5.0",
  "react": "^16.10.1",
  "react-router-dom": "^5.1.2",
}
`;
